test/kafo/hook_context_test.rb

```diff
@@ -43,5 +44,20 @@ module Kafo
         assert_equal context.module_enabled?('known_module'), true
       end
     end
+
+    describe "#app_option?" do
+      app = {:known_option => 'option_arg'}
+      config = Minitest::Mock.new
+      kafo.expect :config, config
+      config.expect :app, app
```

Comment:
AFAIK this is generally a bad practice. The `specify` blocks scope variables to a test. The way you define it here might leak after the test. I'm not that familiar with minitest, but you may be able to use `before {}`.

Reply:
I will add, `expect` is a form of `assertion` and should be used within a `specify` block. You can use `let` to setup the mock itself in spec notation but don't be afraid to repeat yourself in tests -- they can be more verbose as tests should be easy to read and understand. Generally, you use `expect` if you care about the internals that are happening and want to assure they are called to some external API and stubs are used when you need a fake object but aren't trying to assert the internals. Since this test file uses Mocks instead of stubs extensively I think it's OK to continue that pattern.

```
let(:app) { {:known_option => 'option_arg'} }
let(:config) { Minitest::Mock.new }

specify do
  kafo.expect :config, config
  config.expect :app, app
  assert_equal true, context.app_option?('known_option')
end